evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -322,7 +347,131 @@ fn evm_error(retdata: &str) -> Capture<(ExitReason, Vec<u8>), Infallible> {
+                                ExitReason::Succeed(ExitSucceed::Returned),
+                                DUMMY_OUTPUT.to_vec(),
+                            )))
+                        } else {
+                            return ExpectRevertReturn::Create(Capture::Exit((
+                                ExitReason::Succeed(ExitSucceed::Returned),
+                                Some(Address::from_slice(
+                                    &hex::decode("0000000000000000000000000000000000000001")
+                                        .expect("valid address hex")
+                                        [0..20],
+                                )),
+                                Vec::new(),
+                            )))
+                        }
+                    } else if call {
+                        return ExpectRevertReturn::Call(evm_error(&*format!(
+                            "Error != expected error: '{}' != '{}'",
+                            String::from_utf8_lossy(&decoded_data[..]),
+                            String::from_utf8_lossy(&expected_revert)
+                        )))
+                    } else {
+                        return ExpectRevertReturn::Create(evm_create_error(&*format!(
+                            "Error != expected error: '{}' != '{}'",
+                            String::from_utf8_lossy(&decoded_data[..]),
+                            String::from_utf8_lossy(&expected_revert)
+                        )))
+                    }
+                }
+
+                if data == *expected_revert {
+                    if call {
+                        return ExpectRevertReturn::Call(Capture::Exit((
+                            ExitReason::Succeed(ExitSucceed::Returned),
+                            DUMMY_OUTPUT.to_vec(),
+                        )))
+                    } else {
+                        return ExpectRevertReturn::Create(Capture::Exit((
+                            ExitReason::Succeed(ExitSucceed::Returned),
+                            Some(Address::from_slice(
+                                &hex::decode("0000000000000000000000000000000000000001")
+                                    .expect("valid address hex")[0..20],
+                            )),
+                            Vec::new(),
+                        )))
+                    }
+                } else if call {
+                    ExpectRevertReturn::Call(evm_error(&*format!(
+                        "Error data != expected error data: 0x{} != 0x{}",
+                        hex::encode(data),
+                        hex::encode(expected_revert)
+                    )))
+                } else {
+                    ExpectRevertReturn::Create(evm_create_error(&*format!(
+                        "Error data != expected error data: 0x{} != 0x{}",
+                        hex::encode(data),
+                        hex::encode(expected_revert)
+                    )))
+                }
```

Comment:
Can all of this refactored in a helper function to duplicate code with the previous branch? I'd be OK with passing a boolean to trigger a different error message depending on if it's the first or second case, but beyond that they look like the same piece of code?